src/v/coproc/pacemaker.cc

```diff
@@ -94,6 +95,23 @@ ss::future<> pacemaker::stop() {
     co_await _shared_res.transport.stop();
 }
 
+void pacemaker::script_failed_handler(script_id id, std::exception_ptr eptr) {
+    try {
+        std::rethrow_exception(eptr);
+    } catch (const script_failed_exception& e) {
+        vlog(coproclog.error, "Handling script_failed_exception: {}", e.what());
+        vassert(id == e.get_id(), "script_failed_handler id mismatch detected");
+        (void)ss::with_gate(_offs.gate, [this, id] {
+            return container().invoke_on_all([id](pacemaker& p) {
+                return p.remove_source(id).discard_result();
+            });
+        });
+    } catch (const std::exception& e) {
+        vlog(coproclog.error, "Unexpected exception encountered: {}", e.what());
+    }
+    vassert(false, "Unexpected exception encountered");
```

Comment:
does this even get triggered? what is an exception that isn't an std::exception child class, you mean like throwing an integer or smth?

Reply:
in any case, it would not trigger this line of code. it would simply propagate.

Reply:
Yeah its just a catch all for a potential future unexpected result.